### PR TITLE
*: init config for v8.1

### PR DESCRIPTION
**Pull request by Oreoxmt**
### What is changed, added or deleted? (Required)

- update TiDB document links from `dev` to `v8.1`
- update GitHub links from `master` to `release-8.1`
- update hardware-and-software-requirements.md for v8.1
- remove tidb-roadmap.md and update links in TOC

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [x] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/17347
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: releases/release-2.1-ga.md [hunk 1]
```diff
@@ -80,7 +80,7 @@ On November 30, 2018, TiDB 2.1 GA is released. See the following updates in this
 
     - Support Server Side Cursor
 
-    + Add the [HTTP API](https://github.com/pingcap/tidb/blob/master/docs/tidb_http_api.md)
+    + Add the [HTTP API](https://github.com/pingcap/tidb/blob/release-8.1/docs/tidb_http_api.md)
```
Comment (Oreoxmt) [inline releases/release-2.1-ga.md:83]:
```suggestion
    + Add the [HTTP API](https://github.com/pingcap/tidb/blob/release-2.1/docs/tidb_http_api.md)
```

File: releases/release-2.1-ga.md [hunk 2]
```diff
@@ -116,7 +116,7 @@ On November 30, 2018, TiDB 2.1 GA is released. See the following updates in this
 
     - Add the table name and database name in the output information of `admin show ddl jobs`
 
-    - [Support using the `ddl/owner/resign` HTTP interface to release the DDL owner and start electing a new DDL owner](https://github.com/pingcap/tidb/blob/master/docs/tidb_http_api.md)
+    - [Support using the `ddl/owner/resign` HTTP interface to release the DDL owner and start electing a new DDL owner](https://github.com/pingcap/tidb/blob/release-8.1/docs/tidb_http_api.md)
```
Comment (Oreoxmt) [inline releases/release-2.1-ga.md:119]:
```suggestion
    - [Support using the `ddl/owner/resign` HTTP interface to release the DDL owner and start electing a new DDL owner](https://github.com/pingcap/tidb/blob/release-2.1/docs/tidb_http_api.md)
```

File: releases/release-5.3.0.md [hunk 3]
```diff
@@ -75,7 +75,7 @@ In v5.3, the key new features or improvements are as follows:
 - The `GRANT ALL ON performance_schema.*` syntax is no longer permitted. If you execute this statement in TiDB, an error occurs.
 - Fix the issue that auto-analyze is unexpectedly triggered outside the specified time period when new indexes are added before v5.3.0. In v5.3.0, after you set the time period through the `tidb_auto_analyze_start_time` and `tidb_auto_analyze_end_time` variables, auto-analyze is triggered only during this time period.
 - The default storage directory for plugins is changed from `""` to `/data/deploy/plugin`.
-- The DM code is migrated to [the folder "dm" in TiCDC code repository](https://github.com/pingcap/tiflow/tree/master/dm). Now DM follows TiDB in version numbers. Next to v2.0.x, the new DM version is v5.3.0, and you can upgrade from v2.0.x to v5.3.0 without any risk.
+- The DM code is migrated to [the folder "dm" in TiCDC code repository](https://github.com/pingcap/tiflow/tree/release-8.1/dm). Now DM follows TiDB in version numbers. Next to v2.0.x, the new DM version is v5.3.0, and you can upgrade from v2.0.x to v5.3.0 without any risk.
```
Comment (Oreoxmt) [inline releases/release-5.3.0.md:78]:
```suggestion
- The DM code is migrated to [the folder "dm" in TiCDC code repository](https://github.com/pingcap/tiflow/tree/release-5.3/dm). Now DM follows TiDB in version numbers. Next to v2.0.x, the new DM version is v5.3.0, and you can upgrade from v2.0.x to v5.3.0 without any risk.
```

**Review by lilin90 — APPROVED** [review]
/lgtm

**Pull request merged**